### DISCUSSION TITLE
macOS: Restore window buffers when restoring windows after a relaunch

### DISCUSSION THREAD

**Discussion by lilyball** [ghostty-org/ghostty#2679] [Feature Requests, Ideas] [answered]
One of the Terminal.app features I really like is when I quit and relaunch and it restores all my windows, it also restores their buffers. Example screenshot:

<img width="682" alt="screenshot of a Terminal window showing part of a command output followed by a highlighted line that says when it was restored" src="https://github.com/user-attachments/assets/dd81037d-394a-4866-8087-ea35155267b2">

This is handy because it lets me know what I was actually doing in that window, and in particular it often provides a handy reference for the last command I ran if my history has gotten too cluttered to easily find it again (this is very useful in the window I have dedicated to SSHing in to another machine whose hostname I can never fully remember).

Terminal.app actually goes a step further and ships a `/etc/bashrc_Apple_Terminal` file that coordinates with Terminal.app[^1] to implement per-session history. This is something I've always had on my todo list as something to maybe look at reimplementing for Fish if possible. I'm not suggesting that Ghostty attempt to offer the same history restoration thing, though setting/restoring the same `TERM_SESSION_ID` env var Terminal uses might be nice to allow users the option of reimplementing it themselves (or sourcing `/etc/bashrc_Apple_Terminal`, though that also includes the OSC 7 stuff).

[^1]: This is done with a `TERM_SESSION_ID` env var containing a UUID that's unique per window/tab and persists across relaunch.

**Comment by mitchellh** [comment]
This is a duplicate. I’m on mobile so I’ll have to find the other but I know this already exists as an issue or discussion.

**Reply by mitchellh** [reply, marked as answer]
#1847

**Reply by lilyball** [reply]
Ah, I must have been searching for the wrong keywords.
